client/namespace.go

```diff
@@ -0,0 +1,163 @@
+	// in most cases when the table already exists.
+	if gr, err := db.Get(nameKey); err != nil {
+		return err
+	} else if gr.Exists() {
+		return fmt.Errorf("table \"%s\" already exists", desc.Name)
+	}
+
+	ir, err := db.Inc(keys.DescIDGenerator, 1)
+	if err != nil {
+		return err
+	}
+	desc.ID = uint32(ir.ValueInt() - 1)
+
+	return db.Txn(func(txn *Txn) error {
+		descKey := keys.MakeDescMetadataKey(desc.ID)
+		b := &Batch{}
+		b.CPut(nameKey, descKey, nil)
+		b.Put(descKey, &desc)
+		return txn.Commit(b)
+	})
+}
+
+// DescribeTable retrieves the table schema for the specified table.
+func (db *DB) DescribeTable(name string) (proto.TableSchema, error) {
+	schema := proto.TableSchema{}
+	gr, err := db.Get(keys.MakeNameMetadataKey(0, strings.ToLower(name)))
+	if err != nil {
+		return schema, err
+	}
+	if !gr.Exists() {
+		return schema, fmt.Errorf("unable to find table \"%s\"", name)
+	}
+	descKey := gr.ValueBytes()
+	desc := proto.TableDescriptor{}
+	if err := db.GetProto(descKey, &desc); err != nil {
+		return schema, err
+	}
+	if err := proto.ValidateTableDesc(desc); err != nil {
+		return schema, err
+	}
+	schema = proto.TableSchemaFromDesc(desc)
+	return schema, nil
+}
+
+// RenameTable renames a table.
+func (db *DB) RenameTable(oldName, newName string) error {
+	// TODO(pmattis): Should we allow both the old and new name to exist
+	// simultaneously for a period of time? The thought is to allow an
+	// application to access the table via either name while the application is
+	// being upgraded. Alternately, instead of a rename table operation perhaps
+	// there should be a link table operation which adds a "hard link" to the
+	// table. Similar to a file, a table would not be removed until all of the
+	// hard links are removed.
+
+	return db.Txn(func(txn *Txn) error {
+		oldNameKey := keys.MakeNameMetadataKey(0, strings.ToLower(oldName))
+		gr, err := txn.Get(oldNameKey)
+		if err != nil {
+			return err
+		}
+		if !gr.Exists() {
+			return fmt.Errorf("unable to find table \"%s\"", oldName)
+		}
+		descKey := gr.ValueBytes()
+		desc := proto.TableDescriptor{}
+		if err := txn.GetProto(descKey, &desc); err != nil {
+			return err
+		}
+		desc.Name = strings.ToLower(newName)
+		if err := proto.ValidateTableDesc(desc); err != nil {
+			return err
+		}
+		newNameKey := keys.MakeNameMetadataKey(0, desc.Name)
+		b := &Batch{}
+		b.Put(descKey, &desc)
+		b.CPut(newNameKey, descKey, nil)
+		b.Del(oldNameKey)
+		return txn.Commit(b)
+	})
+}
+
+// DeleteTable deletes the specified table.
+func (db *DB) DeleteTable(name string) error {
+	nameKey := keys.MakeNameMetadataKey(0, strings.ToLower(name))
+	gr, err := db.Get(nameKey)
+	if err != nil {
+		return err
+	}
+	if !gr.Exists() {
+		return fmt.Errorf("unable to find table \"%s\"", name)
+	}
+	descKey := gr.ValueBytes()
+	desc := proto.TableDescriptor{}
+	if err := db.GetProto(descKey, &desc); err != nil {
+		return err
+	}
+
+	panic("TODO(pmattis): delete all of the tables rows")
```

Comment:
That's what the design doc says. See the TODO comment in `RenameTable` for my current thoughts on this.